quickshop-bukkit/src/main/java/com/ghostchu/quickshop/util/ItemContainerUtil.java

```diff
@@ -0,0 +1,49 @@
+package com.ghostchu.quickshop.util;
+
+import org.bukkit.block.Container;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.BlockStateMeta;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A simple utility class for handling container items such as Shulker Boxes.
+ * @author TauCubed
+ */
+public class ItemContainerUtil {
+
+    /**
+     * Checks if the provided ItemStack is a container, if so, flattens the contents of the container
+     * @param stack the stack to check
+     * @param recursive should the check function recursively? (chests inside chests inside che...)
+     * @param includeInput should the input ItemStack be included in the results?
+     * @return a list of flattened contents
+     */
+    public static List<ItemStack> flattenContents(ItemStack stack, boolean recursive, boolean includeInput) {
+        List<ItemStack> flat = new ArrayList<>();
+        if (includeInput) {
+            flat.add(stack);
+        }
+
+        flattenContents0(flat, stack, recursive);
+
+        return flat;
+    }
+
+    private static void flattenContents0(List<ItemStack> flat, ItemStack stack, boolean recursive) {
+        if (stack.getItemMeta() instanceof BlockStateMeta meta
+                && meta.hasBlockState()
+                && meta.getBlockState() instanceof Container container) {
```

Comment:
Is there a reason for the inclusion of hasBlockState before instanceof? If it's for null checking, that is built into instanceof and can be removed.

Reply:
getBlockState doesn't return null. It creates a new block state if one doesn't exist. Most items (even if they they have a block state when placed) don't have a block state attached to their item.

This check is just for efficiency as it is very light weight.

Reply:
Fair enough.